### PR TITLE
New L1REPACK workflow for DATA with simulation of TPs (81x)

### PR DESCRIPTION
**Pull request by rekovic**
This is the 81x version of 80x #15117
This super-seeds #15118 as it resolves its merge conflicts into CMSSW_8_1_X.

For completion, here are the comments from #15118:

This is a test.

A new L1T workflow defined as step `L1REPACK:FullSimTP` for running on DATA is introduced.
It is designed for the needs of AlCa group for tests at RelVal where most of Trigger Primitives are re-simulated and used as inputs to L1T.  

In this workflow L1T emulation uses the following TPs:
-- sim DT TPs
-- unpacked CSC TPs 
-- sim Ecal TPs
-- sim Hcal TPs  

Note: Unpacked CSC TPs are used because the simCSC TPs have offset of BX=2.  Till this is fixed, 
prefer to use unpacked CSC TPs than to individually configure muon track finders (EMTF, OMTF) emulators to do internal BX shifting of its input TPs.

An example command using the step `L1REPACK:FullSimTP` on a RAW 2016 data file:

<pre>
cmsDriver.py L1TEST --python_filename=l1t_test.py -s L1REPACK:FullSimTP  --era=Run2_2016 --conditions=80X_dataRun2_HLT_v12 -n 10 --data --filein=/store/data/Run2016B/ZeroBias/RAW/v2/000/274/442/00000/F08D3D87-762C-E611-BBB8-02163E014160.root --eventcontent  FEVTDEBUGHLT  
</pre>

### CODE REVIEW

**Comment by fabozzi** [comment]
please test

File: Configuration/PyReleaseValidation/python/relval_steps.py [hunk 1]
```diff
@@ -1162,6 +1164,7 @@ def lhegensim(fragment,howMuch):
 steps['RECODR2_25nsreHLT']=merge([{'--hltProcess':'reHLT'},steps['RECODR2_25ns']])
 steps['RECODR2_50nsreHLT']=merge([{'--hltProcess':'reHLT'},steps['RECODR2_50ns']])
 steps['RECODR2_2016reHLT']=merge([{'--hltProcess':'reHLT','--conditions':'auto:run2_data_relval'},steps['RECODR2_2016']])
+steps['RECODR2newL1repack_2016reHLT']=merge([{'-s':'L1REPACK:FullSimTP,RAW2DIGI,L1Reco,RECO,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@miniAODDQM','--hltProcess':'reHLT'},steps['RECODR2_2016']])
```
Comment (rekovic) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: As this is a rebase of 80x PR, the above line does not use ```'--conditions':'auto:run2_data_relval'```
which in the meantime has been used in 7c6f44809f8bd39dec429a04ec588f967d3bee27 to line above.
@mmusich If you confirm, I can change the above line to use it, as: 
<pre>
+steps['RECODR2newL1repack_2016reHLT']=merge([{'-s':'L1REPACK:FullSimTP,RAW2DIGI,L1Reco,RECO,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@miniAODDQM','--hltProcess':'reHLT','--conditions':'auto:run2_data_relval'},steps['RECODR2_2016']])
</pre>
            
Mind you that ```'--conditions':'auto:run2_data_relval'``` is not used in the 80x PR #15117 [here](https://github.com/cms-sw/cmssw/pull/15117/files#diff-819af2ec677272fddf1fef50e788a88bR1142)
and if needed I can update it there also.  Let me know.

Comment (mmusich) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: @rekovic:
```
cmsrel CMSSW_8_1_X_2016-07-12-2300
cd CMSSW_8_1_X_2016-07-12-2300/src/
cmsenv
git cms-merge-topic 15182
scramv1 b -j 8
runTheMatrix.py -nel 136.7321
```
gives me:
```
 [1]: input from: /ZeroBias/Run2016B-v2/RAW with run [] 
 [2]: cmsDriver.py step2  --datatier FEVTDEBUGHLT --conditions auto:run2_hlt_relval -s L1REPACK:FullSimTP,HLT:@relval2016 --process reHLT --data  --era Run2_2016 --eventcontent FEVTDEBUGHLT -n 100 
 [3]: cmsDriver.py step3  --conditions auto:run2_data_relval -s L1REPACK:FullSimTP,RAW2DIGI,L1Reco,RECO,EI,PAT,ALCA:SiStripCalZeroBias+SiStripCalMinBias+TkAlMinBias+EcalESAlign,DQM:@standardDQM+@miniAODDQM --runUnscheduled  --process reRECO --data  --era Run2_2016 --eventcontent RECO,MINIAOD,DQM --hltProcess reHLT --scenario pp --datatier RECO,MINIAOD,DQMIO --customise Configuration/DataProcessing/RecoTLR.customisePostEra_Run2_2016 -n 100 
 [4]: cmsDriver.py step4  --data  --filetype DQM --conditions auto:run2_data_relval -s HARVESTING:@standardDQM+@miniAODDQM --scenario pp -n 100 
```
just to be clear, the worry is that the `L1REPACK:FullSimTP` in step2 and step3 are consuming different Global Tags?

**Comment by franzoni** [comment]
please test

File: Configuration/PyReleaseValidation/python/relval_steps.py (hunk 1 above)
Comment (rekovic) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: @mmusich Not exactly.  My worry was  that the step3 defined for a workflow ```'RECODR2_2016reHLT'``` defined above as:
<pre>
 steps['RECODR2_2016reHLT']=merge([{'--hltProcess':'reHLT','--conditions':'auto:run2_data_relval'},steps['RECODR2_2016']])
</pre>
had explicitly used ```'--conditions':'auto:run2_data_relval'``` in the definition, while that was not the case for the workflow ```'RECODR2newL1repack_2016reHLT'```.

Judging by the output of your command ```runTheMatrix.py -nel 136.7321``` the same flag  ```'--conditions':'auto:run2_data_relval'``` is picked up in step3 for the later workflow, so I guess we are fine.

Comment (fabozzi) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: @rekovic 
steps['RECODR2_2016'] contains already '--conditions':'auto:run2_data_relval'

https://github.com/cms-l1t-offline/cmssw/blob/a2b4491746fdd457a8fffc51395a36b9d900004e/Configuration/PyReleaseValidation/python/relval_steps.py#L1030

In steps['RECODR2newL1repack_2016reHLT'] it is not overwritten in the merge command, so we have the same conditions as in steps['RECODR2_2016'].
This is OK, I'ld say.

Comment (mmusich) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: Indeed that's OK with me as well!

**Comment by franzoni** [comment]
please test

File: Configuration/PyReleaseValidation/python/relval_steps.py (hunk 1 above)
Comment (rekovic) [inline Configuration/PyReleaseValidation/python/relval_steps.py]: @fabozzi @mmusich  Thanks.

**Comment by fabozzi** [comment]
please test

**Comment by fabozzi** [comment]
+1

**Comment by davidlange6** [comment]
@rekovic - this needs a rebase

**Pull request closed without merging**